Zotero/Scenes/Detail/PDF/Views/PDFAnnotationsViewController.swift

```diff
@@ -153,124 +157,121 @@ final class PDFAnnotationsViewController: UIViewController {
+                    snapshot.reloadSections([0])
+                    dataSource.apply(snapshot, animatingDifferences: isVisible, completion: completion)
                 }
-                dataSource.apply(snapshot, animatingDifferences: isVisible, completion: completion)
+                return
             }
-            return
-        }
 
-        if state.changes.contains(.interfaceStyle) {
-            updateQueue.async { [weak self] in
-                guard let self else { return }
-                var snapshot = dataSource.snapshot()
-                guard !snapshot.sectionIdentifiers.isEmpty else { return }
-                snapshot.reloadSections([0])
-                dataSource.apply(snapshot, animatingDifferences: false, completion: completion)
+            if state.changes.contains(.annotations) {
+                updateUI(state: state, animatedDifferences: isVisible, completion: completion)
+                return
             }
-            return
-        }
 
-        if state.changes.contains(.selection) || state.changes.contains(.activeComment) {
-            if let keys = state.updatedAnnotationKeys {
-                updateQueue.sync { [unowned self] in
+            if state.changes.contains(.interfaceStyle) && dataSource.snapshot().numberOfItems > 0 {
+                updateQueue.async { [weak self] in
+                    guard let self else { return }
                     var snapshot = dataSource.snapshot()
-                    snapshot.reloadItems(keys)
-                    dataSource.apply(snapshot, animatingDifferences: false)
+                    snapshot.reconfigureItems(snapshot.itemIdentifiers)
+                    dataSource.apply(snapshot, animatingDifferences: false, completion: completion)
                 }
+                return
             }
 
-            updateCellHeight()
-            focusSelectedCell()
-
-            if state.changes.contains(.sidebarEditing) {
-                tableView.setEditing(state.sidebarEditingEnabled, animated: isVisible)
+            if state.changes.contains(.selection) || state.changes.contains(.activeComment) {
+                var keys = state.updatedAnnotationKeys ?? []
+                if let key = state.selectedAnnotationKey, !keys.contains(key) {
+                    keys.append(key)
+                }
+                if !keys.isEmpty {
+                    updateQueue.async { [weak self] in
+                        guard let self else { return }
+                        var snapshot = dataSource.snapshot()
+                        snapshot.reconfigureItems(keys)
+                        dataSource.apply(snapshot, animatingDifferences: false) { [weak self] in
+                            guard let self else { return }
+                            focusSelectedCell()
+                            completion()
+                        }
+                    }
+                } else {
```

Comment:
Just a nitpick, but instead of `else` you could just return after `if`, completion will be called below.

Reply:
Check https://github.com/zotero/zotero-ios/pull/1040/commits/ffd70d194f89db903d15f860defbdc0edf565f17